docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,35 @@
+# Architecture Overview
+
+In the following the architecture of PowerDynamics with its components is explained in more detail.
+
+## Components of PowerDynamics
+
+The main components of PowerDynamics are 
+- nodes
+- lines
+- PowerGrid
+- operationpoint
+- faults
+- PowerGridSolutions
+
+The relationships between the components is illustrated in the figure below:
+
+<img src="figures/PowerDynamics_Architecturemd.svg">
```

Comment:
some of the nodes are types some are abstract concepts. maybe all of them could be the PD-jl typs?

Reply:
I don't know what you mean. Do you want to change the picture or the general concept of PD?

Concerning your changes: the faults alter the PowerGrid. That is why I let their arrow end at this component instead of operationpoint.

Reply:
What I meant above is that there is a problem on how to interpret such a diagam. Is it supposed to tell the user how the defined type and functions interact? Or should it give a somehow general workflow scheme? Both is nice to have but now it is kind of mixed, causing confusion about what the arrows mean (and whether there are different kinds of arrows).

Reply:
Maybe I ask @janlisse  once more since he comes from the Software development side. The idea was not to present a workflow itself but what are the components and how do they build new components. An easy one is node and lines build the power grid. But yes, it somehow ended in something that looks like a workflow.